### DISCUSSION TITLE
Why does IPhython quit on `Ctrl+C` (MacOS)?

### DISCUSSION THREAD

**Discussion by VladimirFokow** [3b1b/manim#2236] [Q&A]
Hello,

On MacOS:
after we start `manimgl`,
and run some long-running command in the IPython terminal -
if we want to cancel it (while it's executing) - we press `Ctrl+C`
but then the whole IPython terminal quits.

For example:

```bash
manimgl
```

```py
from time import sleep
sleep(5)  # Ctrl+C   ---> it quits manimgl, but it shouldn't
```

<br />

Is it possible to just cancel the command 
(and NOT quit the manimgl's IPython terminal)?

On Linux it works (the IPython terminal doesn't quit on `Ctrl+C`). 
Why is it different on MacOS?

<br /><br />

Note:

- Is [this code](https://github.com/3b1b/manim/blob/1a65498f977aab7ed505fa7bb660399fa64c19af/manimlib/scene/scene.py#L227) relevant?
- in normal IPython, `Ctrl+C` sends `KeyboardInterrupt` which cancels execution withOUT quitting the IPython terminal.

<br />

(manimgl version used: 1.7.1 - currently latest)
Thank you for any suggestions!

**Comment by Splines** [comment]
To check that this is actually a Manim issue and not one from an IPython shell, could you @VladimirFokow and @bhoov check on MacOS if the following works for you?

Use this Python code that is not related to Manim at all:

```py
from IPython.terminal.embed import InteractiveShellEmbed


class MyMainClass:
    def main(self):
        shell = InteractiveShellEmbed()
        shell()


if __name__ == "__main__":
    main_class = MyMainClass()
    main_class.main()

```

Execute it, then in the spawned IPython shell, run `import time; time.sleep(10)`. While we sleep, try to `Ctrl/Cmd+C` cancel the interactive shell. For me on Linux, it just interrupts the current cell execution *and not* the whole IPython kernel. Is this different for you?
